src/averageAges.js

```diff
@@ -37,9 +44,23 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
-}
+  const women = withChildren
+    ? people.filter(person =>
+      person.mother !== null).map(person => person.mother
+    )
+    : people.filter(person => person.sex === 'f');
+
+  const mothers = people.filter(person => women.includes(person.name));
+  const motherAge = mothers.map(person => person.died - person.born);
+
+  const allAge = women.map(woman => woman.died - woman.born);
+  const sum = withChildren
+    ? (motherAge.reduce((prev, next) =>
+      prev + next) / motherAge.length).toFixed(2)
+    : (allAge.reduce((prev, next) => prev + next) / allAge.length).toFixed(2);
```

Comment:
Create a separate function for this & reuse in all cases
```suggestion
    ? getAvarageAge(motherAge)
    : getAvarageAge(allAge);
```